pkg/doc/jose/jwk.go

```diff
@@ -24,101 +24,211 @@ import (
 )
 
 const (
-	secp256k1Crv = "secp256k1"
-	secp256k1Kty = "EC"
-	bitsPerByte  = 8
+	secp256k1Alg  = "ES256K"
+	secp256k1Crv  = "secp256k1"
+	secp256k1Kty  = "EC"
+	secp256k1Size = 32
+	bitsPerByte   = 8
 )
 
 // JWK (JSON Web Key) is a JSON data structure that represents a cryptographic key.
-type JWK jose.JSONWebKey
+type JWK struct {
+	jose.JSONWebKey
 
-// JWK gets JWK from JOSE headers.
-func (h Headers) JWK() (*JWK, bool) {
-	jwkRaw, ok := h[HeaderJSONWebKey]
-	if !ok {
-		return nil, false
-	}
-
-	var jwk JWK
+	x   *big.Int
+	y   *big.Int
+	Kty string
+	Crv string
+}
```

Comment:
there's no `kid` needed?

Reply:
It's already there, inside embedded `jose.JSONWebKey`. `x`, 'y', 'key type', 'curve' are not copied from `jose.rawJSONWebKey` to `jose.JSONWebKey` as a result of unmarshalling. That's why I had to add them to `JWK` struct to support `secp256k1` properly.

Reply:
there seem to be duplication of fields with JSONWebKey then.. it's better to remove duplicate fields